### ISSUE TITLE
Generate a Request upon submission instead of after sending it to an agency component

### ISSUE THREAD

**Issue by malikkotob** [18F/beta.foia.gov#215]
As an agency component, in order to track down a request, I need some sort of request ID to be sent by the portal with every request.

[Description, background context, or implementation details]

## Acceptance criteria
- [x] Upon submission, a FOIA Request entity is created that references the following information:
    - webform submission ID
    - time submission was received
    - requester email address (if available)
- [x] The FOIA Request entity ID is queued instead of the webform submission ID
- [x] The FOIA Request entity is updated to have a status of `queued`
- [x] The Request ID is made available to both the API and Email sending mechanisms
- [x] The API sending service makes use of the newly available Request ID by including it in the JSON under the key `request_id`

## Technical Requirements

In `FoiaSubmissionQueueHandler.php`:
- [x] A `createRequest` method is added which creates a FOIA Request entity (#146) that:
    - references the webform submission ID
    - includes the time the submission was received
    - includes the requester email address (if available)
- [x] The `queueSubmission` method is updated to be called `queueRequest`
- [x] The `postSave` method is updated to:
    - [x] call `createRequest`
    - [x] pass the created request to `queueRequest`
    - [x] update the status on the request entity to "queued" or something similar (see #146 for statuses)

In `FoiaSubmissionQueueWorker.php`
- [x] The Request entity or Request ID should be passed to the `sendSubmissionToComponent` method, along with the existing parameters

In `FoiaSubmissionServiceApi.php`
- [x] The `assembleRequestData` method updated to include the Request entity id in a new key called `request_id`

## Functional Review

- [x] Login to dev
- [x] Go to `/form/demo-form-one`
- [x] Submit the form with fields filled out (including the email field)
- [x] Go to `/admin/content/foia-requests`
- [x] Click on the FOIA request corresponding to your submission
- [x] Verify the email address you entered is present under `Requester Email`
- [x] Verify that the webform submission ID is present under `Webform Submission ID`
- [x] Go to `/form/demo-form-one`
- [x] Submit the form without the email field
- [x] Go back to `/admin/content/foia-requests`
- [x] Click on the FOIA request corresponding to your submission
- [x] Verify there is NO email address present under `Requester Email`
- [x] Verify that the webform submission ID is present under `Webform Submission ID`
- [x] Go to `/node/4901/edit`
- [x] Scroll down to the 'Request Submission Form' field
- [x] Deselect the `Demo Form One` form (you can either associate the component with another form or disassociate the component from any form)
- [x] Go to `/form/demo-form-one`
- [x] Submit the form
- [x] Go back to `/admin/content/foia-requests`
- [x] Verify there is NO new foia request in the table

## Developer Review
- [x] Review [PR 98](https://github.com/usdoj/foia-api/pull/98)

*Setup*
- [x] Run `blt sync:db`
- [x] Checkout `doj-215-mk` locally
- [x] Clear caches: `drush cr`

*Testing*
- [x] Login and navigate to http://local.dojfoia.gov/node/4901/edit
- [x] Update the 'Portal Submission Format' to be 'API'
- [x] Update the 'Submission API URL' field to have the value `https://example.com`
- [x] Update the 'Submission API Shared Secret' field to have the value `12345`
- [x] Go to http://local.dojfoia.gov/admin/structure/webform/manage/demo_form_one
- [x] Edit the email field to no longer be required
- [x] Complete functional review steps

*Automated Testing*
- [x] Run `blt tests:all` and verify that all tests are passing

*Setup on dev environment*
- [x] Ping Malik to setup on dev, including making the email field not required on the form

**Comment by malikkotob** [comment]
@adborden I just want to call out that the current plan here will allow us to send a unique request ID to the receiving agency component. It also paves the way for being able to send the same ID to a requester in an e-mail, if that functionality comes out as a need in V2.

**Comment by malikkotob** [comment]
@adborden Yes, the plan is to include this in emails as well. The AC above mentions:
> The Request entity or Request ID should be passed to the `sendSubmissionToComponent` method, along with the existing parameters

This method belongs to an interface to be implemented by both our submission API class, as well as our submission email class.

**Comment by adborden** [comment]
Great! Do you mind if I add these AC in plain English so clear to anyone doing acceptance? The current AC is very implementation specific.
- [ ] request submission payload to agency components by API includes unique request ID
- [ ] email submissions to agency components include unique request ID

**Comment by malikkotob** [comment]
@adborden I split things up into technical requirements and acceptance criteria both here and in #122. In this story, we'll limit the scope to making the request ID available to the email service. I updated #122 to actually make use of the ID within the email body.

**Comment by adborden** [comment]
Works for me, thanks!

**Comment by malikkotob** [comment]
Merged into `develop`
